comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,42 +10,130 @@ public enum Mark { None, O, X };
 
 	public class OxoBoard
 	{
+        Mark[,] borad;
 		// Constructor. Perform any necessary data initialisation here.
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		public OxoBoard( int width = 3, int height = 3, int inARow = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            borad = new Mark[width,height];
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            return borad[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			if(x < borad.GetLength(0) && y < borad.GetLength(1))
+            {
+                if (borad[x,y] == Mark.None)
+                {
+                    borad[x, y] = mark;
+                    return true;
+                }
+            }
+
+            return false;
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            for (int p = 0; p < borad.GetLength(0); p++)
+            {
+                for (int j = 0; j < borad.GetLength(1); j++)
+                {
+                    if (borad[p,j] == Mark.None)
+                    {
+                        return false;
+                    }
+                }
+            }
+            return true;
 		}
 
 		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
 		// Otherwise, return Mark.None.
 		public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			foreach(Mark mark in new Mark [Mark.O, Mark.X])
+                {
+                    if (DiagonalCheck1(mark) || DiagonalCheck2(mark))
+                {
+                    return mark;
+                }
```

Comment:
You seem to be missing a closing brace here -- this is causing a number of compile errors later on